### PR TITLE
Fix: extend Cassandra e2e test timeouts

### PR DESCRIPTION
**Pull request by nilayasiktoprak**
Signed-off-by: nilayasiktoprak <nilayasiktoprak@gmail.com>

Extend Cassandra e2e test timeouts

### Checklist

- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Relates #2227

### CODE REVIEW

File: tests/scalers/cassandra.test.ts [hunk 1]
```diff
@@ -152,7 +162,7 @@ spec:
         app: cassandra-app
     spec:
       containers:
-      - image: bitnami/cassandra:4.0.1
```
Comment (zroubalik) [inline tests/scalers/cassandra.test.ts:155]: I did this change, because bitnami/cassandra is non-root, so please revert this change.

File: tests/scalers/cassandra.test.ts [hunk 2]
```diff
@@ -52,15 +57,17 @@ test.before(t => {
...
     fs.writeFileSync(nginxTmpFile.name, nginxDeployYaml)
 
     t.is(0, sh.exec(`kubectl apply --namespace ${cassandraNamespace} -f ${nginxTmpFile.name}`).code, 'creating nginx deployment should work.')
     // wait for nginx to load
+    console.log("wait for nginx to load")
     let nginxReadyReplicaCount = '0'
      for (let i = 0; i < 30; i++) {
         nginxReadyReplicaCount = sh.exec(`kubectl get deploy/${nginxDeploymentName} -n ${cassandraNamespace} -o jsonpath='{.status.readyReplicas}'`).stdout
-        if (nginxReadyReplicaCount != '1') {
+        if (nginxReadyReplicaCount != '') {
```
Comment (zroubalik) [inline tests/scalers/cassandra.test.ts:70]: Don't follow this change?

File: tests/scalers/cassandra.test.ts [hunk 3]
```diff
@@ -43,6 +47,7 @@ test.before(t => {
     t.is('1', cassandraClientReadyReplicaCount, 'Cassandra client is not in a ready state')
 
     // create table
+    console.log("create table")
```
Comment (zroubalik) [inline tests/scalers/cassandra.test.ts:50]: As I mentioned on the issue, somewhere around here should be some checks, because even if the cassandra and cassandra client pods are ready, the Cassandra itself is not fully started. So would be nice to run some CQLquery here in wait loop, to check once cassandra is up. The query could just check status or something.

File: tests/scalers/cassandra.test.ts (hunk 2 above)
Comment (nilayasiktoprak) [reply tests/scalers/cassandra.test.ts:70]: It starts with 0 replica at first, not 1, so this check was wrong

Comment (JorTurFer) [reply tests/scalers/cassandra.test.ts]: In that case, should it be different from 0 instead of empty?

File: tests/scalers/cassandra.test.ts [hunk 4]
```diff
@@ -12,57 +12,90 @@ const nginxDeploymentName = 'nginx-deployment'
 
 test.before(t => {
     // install cassandra
+    console.log("Install cassandra")
```
Comment (JorTurFer) [inline tests/scalers/cassandra.test.ts:15]: I know that in one or two scalers we have some console.log, but, do they make sense? 🤔

Comment (nilayasiktoprak) [reply tests/scalers/cassandra.test.ts:15]: After all those fails, I thought it might be good to put some logs to detect where exactly the test stucks. I can remove them if you wish

**Comment by nilayasiktoprak** [comment]
> /run-e2e cassandra*

So it passed?? 🤩

File: tests/scalers/cassandra.test.ts (hunk 2 above)
Comment (nilayasiktoprak) [reply tests/scalers/cassandra.test.ts]: I'm not sure if I got your question right, so do you mean it should be like this:
`if (nginxReadyReplicaCount != '0') {`

When I stdout the nginxReadyReplicaCount at that stage, it prints empty, that's why I changed it this way. However, I don't think that checking a deployment with 0 replica at the beginning whether it is empty or zero makes sense. This control is needless, I guess. WDYT?

Comment (JorTurFer) [reply tests/scalers/cassandra.test.ts]: It depends on the test itself. I mean, if the test check that the scaler scales from 0 to X, IMHO it's a good idea checking that the initial state is really 0 instead of any other initial state that could be produced by a side effect. 
In general (IMO again), always it's good idea during the tests to assert the initial state. 
Any opinion here @zroubalik ?

BTW, you could use the existing helper to check the current amount of instances instead of querying directly. It is easier to read and maintain. [This is a usage example](https://github.com/kedacore/keda/blob/main/tests/scalers/rabbitmq-queue-http.test.ts#L36)

File: tests/scalers/cassandra.test.ts (hunk 4 above)
Comment (JorTurFer) [reply tests/scalers/cassandra.test.ts:15]: No no, I don't have any strong opinion about this, I was just sharing my thoughts :)

**Comment by JorTurFer** [comment]
> > /run-e2e cassandra*
> 
> So it passed?? 🤩

Yes! The test looks good! 🥳 🎉  (There is only pending the open conversations :) )
Thanks for your contribution!!!!

**Review by zroubalik — APPROVED** [review]

**Comment by zroubalik** [comment]
Looking good, let's merge it :)
